nucypher/device/trezor.py

```diff
@@ -0,0 +1,178 @@
+
+    @_handle_device_call
+    def sign_message(self, message: bytes, checksum_address: str):
+        """
+        Signs a message via the TREZOR ethereum sign_message API and returns
+        the signature and the address used to sign it. This method requires
+        interaction between the TREZOR and the user.
+
+        It will sign the message with the account described by checksum_address.
+        """
+        try:
+            bip44_path = self.__addresses[checksum_address]
+        except KeyError:
+            raise self.DeviceError(f'{checksum_address} is not available as a cached address on this device.')
+
+        sig = trezor_eth.sign_message(self.client, bip44_path, message)
+        return self.Signature(sig.signature, sig.address)
+
+    @_handle_device_call
+    @validate_checksum_address
+    def verify_message(self, signature: bytes, message: bytes, checksum_address: str):
+        """
+        Verifies that a signature and message pair are from a specified
+        address via the TREZOR ethereum verify_message API. This method
+        requires interaction between the TREZOR and the user.
+
+        If the signature or message is not valid, it will raise a
+        nucypher.crypto.signing.InvalidSignature exception. Otherwise, it
+        will return True.
+        """
+        is_valid = trezor_eth.verify_message(self.client, checksum_address,
+                                             signature, message)
+        if not is_valid:
+            raise InvalidSignature("Signature verification failed.")
+        return True
+
+    @_handle_device_call
+    def get_address(self, address_index: int = None, hd_path: str = None):
+        """
+        Derives an address available on the Trezor via the ethereum
+        get_address API and returns it.
+        """
+        if address_index is not None and hd_path is None:
+            hd_path = self.ETH_BIP44_PATH.format(address_index=address_index)
+            hd_path = trezor_tools.parse_path(hd_path)
+        elif address_index is None and hd_path is not None:
+            hd_path = trezor_tools.parse_path(hd_path)
+        else:
+            raise ValueError("You must provider either an address_index or an hd_path.")
+
+        address = trezor_eth.get_address(self.client, hd_path)
+        return address
+
+    @_handle_device_call
+    def sign_eth_transaction(self, checksum_address: str, transaction: dict) -> bytes:
+        """
+        Signs an Ethereum transaction via the Trezor ethereum sign_tx API
+        and returns the signed, RLP encoded transaction.
+
+        TODO: Is there any input validation required for the transaction
+              data that is passed in?
+        """
+        try:
+            bip44_path = self.__addresses[checksum_address]
+        except KeyError:
+            raise self.DeviceError(f'{checksum_address} is not available as a cached address on this device.')
+
+        # Handle Web3.py -> trezor API formatting
```

Comment:
There is an additional transaction formatting step that needs to be performed here, since incoming transaction data is a hex `str`, but the `sign_tx` interface expects `bytes`.: 

```
        if unsigned_transaction['data']:
            unsigned_transaction['data'] = Web3.toBytes(hexstr=unsigned_transaction['data'])
```